### PR TITLE
Do not close all streams right after container stopped

### LINKED ISSUE #604: container stop may interrupt ongoing std streams
Illustrated as the following figure, the container stop event will lead to streamconfig.CloseStreams(), which will close stdin pipe, and then lead to tty.Close()

img_4285

This issue was found during process #586 and has not fixed yet.

### PR DESCRIPTION
**Pull request by gnawux**
fix #604

The container stopping should hup all streams if there are any.
Just check and reclaim any remains after a period should be safe.

Need update #585 later if this is approved

Signed-off-by: Wang Xu <gnawux@gmail.com>

### CODE REVIEW

**Comment by gnawux** [comment]
I did more than 100 test and it did not interrupt output any more. let's see if it affects other cases.

**Comment by gnawux** [comment]
travis passed

hykins failed because of #601 , please retest it

**Comment by gnawux** [comment]
retest it please, hykins

**Comment by gnawux** [comment]
retest this please, hykins

**Comment by gnawux** [comment]
hykins failed again because of #601, retest this please

**Comment by Crazykev** [comment]
I test this with command `hyperctl run --rm busybox echo hello`, will delay 3 seconds to return, is this expected side-effect?

**Comment by gnawux** [comment]
@Crazykev I didn't found this and will try to reproduce it.

**Comment by gnawux** [comment]
I extends the timeout from 3 seconds to 10 seconds and confirm what you report. Will check and fix it.

**Comment by gnawux** [comment]
The reason is `stdout` and `stderr` of `TtyIO` are not have to be a `Closer`, esp. the stdout/stderr of StreamConfig and stdcopy streams are not `Closer`.

**Comment by gnawux** [comment]
tested with `run` and `exec`, both work

**Comment by gnawux** [comment]
retest this please, hykins

**Comment by gnawux** [comment]
retest this please, hykins

File: daemon/pod/exec.go [hunk 1]
```diff
@@ -78,6 +78,15 @@ func (wc *waitClose) Close() error {
 	return wc.ReadCloser.Close()
 }
 
+type writeCloser struct {
+	io.Writer
+	closer io.Closer
+}
```
Comment (laijs) [inline daemon/pod/exec.go]:
```diff
 +type writeCloser struct {
 +	io.Writer
 +	io.Closer
 +}
```

**Pull request merged**